src/compiler/types.ts

```diff
@@ -9850,6 +9852,34 @@ export const commentPragmas = {
     },
 } as const;
 
+export const enum JSDocParsingMode {
+    /**
+     * Always parse JSDoc comments and include them in the AST.
+     *
+     * This is the default if no mode is provided.
+     */
+    ParseAll,
+    /**
+     * Never parse JSDoc comments, mo matter the file type.
+     */
+    ParseNone,
+    /**
+     * Parse only JSDoc comments which are needed to provide correct type errors.
+     *
+     * This will always parse JSDoc in non-TS files, but only parse JSDoc comments
+     * containing `@see` and `@link` in TS files.
+     */
+    ParseForTypeErrors,

```

Comment:
Last minute bikeshedding; `ParseForErrors`? Technically, this covers grammar checks.

Reply:
But, not actually sure we do grammar checks which relate to JSDoc.

Reply:
Not in TS files, no. Just in JS files.